File: frontend/lib/models/campagne.dart
class Campagne {
  final int id;
  final String nom;
  final String? description;
  final String? zone;
  final bool estCloturee;
  final String dateCreation;
  final int createurId;
  final String? createurNom;
  final int nombreFuites;

  Campagne({
    required this.id,
    required this.nom,
    this.description,
    this.zone,
    required this.estCloturee,
    required this.dateCreation,
    required this.createurId,
    this.createurNom,
    this.nombreFuites = 0,
  });

  factory Campagne.fromJson(Map<String, dynamic> json) {
    return Campagne(
      id: json['id'] as int,
      nom: json['nom'] as String,
      description: json['description'] as String?,
      zone: json['zone'] as String?,
      estCloturee: json['estCloturee'] as bool,
      dateCreation: json['dateCreation'] as String,
      createurId: json['createurId'] as int,
      createurNom: json['createurNom'] as String?,
      nombreFuites: json['nombreFuites'] as int? ?? 0,
    );
  }
}

File: backend/src/main/java/com/backend/backend/dao/entities/Campagne.java
package com.backend.backend.dao.entities;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.util.Date;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "campagnes")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Campagne {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String nom;

    private String description;

    private String zone;

    @Column(nullable = false)
    private Boolean estCloturee = false;

    @Temporal(TemporalType.TIMESTAMP)
    @Column(nullable = false)
    private Date dateCreation;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "utilisateur_id", nullable = false)
    private Utilisateur createur;

    @OneToMany(mappedBy = "campagne", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Fuite> fuites = new ArrayList<>();
}
